core/Piranha.Manager.Core/Services/MediaService.cs

```diff
@@ -79,36 +79,25 @@ public async Task<MediaListModel> GetList(Guid? folderId = null, MediaType? filt
                 }
             }
 
-            model.Media = (await _api.Media.GetAllAsync(folderId))
-                .Select(m => new MediaListModel.MediaItem
+            var holdMedia = (await _api.Media.GetAllByIdAsync(folderId));
+            if (filter.HasValue)
+            {
+                holdMedia = holdMedia
+                    .Where(m => m.Type == filter.Value);
+            }
+            var pairMedia = holdMedia.Select(m => new { media = m, mediaItem = new MediaListModel.MediaItem
                 {
                     Id = m.Id,
                     FolderId = m.FolderId,
                     Type = m.Type.ToString(),
                     Filename = m.Filename,
-                    PublicUrl = m.PublicUrl.Replace("~", ""),
+                    PublicUrl = m.PublicUrl.TrimStart('~'), //Will only enumerate the start of the string, probably a faster operation.
                     ContentType = m.ContentType,
                     Size = Utils.FormatByteSize(m.Size),
                     Width = m.Width,
                     Height = m.Height,
                     LastModified = m.LastModified.ToString("yyyy-MM-dd")
-                }).ToList();
-
-            if (filter.HasValue)
-            {
-                model.Media = model.Media
-                    .Where(m => m.Type == filter.Value.ToString())
-                    .ToList();
-            }
-
-            if (model.Media.Where(m => m.Type == "Image").Count() > model.Media.Count / 2)
-            {
-                model.ViewMode = MediaListModel.GalleryView;
-            }
-            else
-            {
-                model.ViewMode = MediaListModel.ListView;
-            }
```

Comment:
Why has this code been removed? This selects how the data should be displayed in the manager depending on if more than 50% of the items are images.

Reply:
Never mind, I actually found the code further down :)